daemon/algod/api/server/v2/handlers.go

```diff
@@ -2044,10 +2103,19 @@ func (v2 *Handlers) GetTransactionGroupLedgerStateDeltasForRound(ctx echo.Contex
 	if err != nil {
 		return notFound(ctx, err, fmt.Sprintf(errFailedRetrievingStateDelta, err), v2.Log)
 	}
-	response := struct {
-		Deltas []eval.TxnGroupDeltaWithIds
-	}{
-		Deltas: deltas,
+	var response interface{}
+	if handle == protocol.JSONStrictHandle {
+		response = struct {
+			Deltas []TxnGroupDeltaWithIDsJSONSerializable
+		}{
+			Deltas: convertTxnGroupDeltasWithIDs(deltas),
+		}
+	} else {
```

Comment:
is there a test case to be added for this?

Reply:
Yes, the modified e2e tests check the JSON output